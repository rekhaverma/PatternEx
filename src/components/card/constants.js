export const COUNT = 'count';
export const TIMESTAMP = 'timestamp';
export const MORE = 'more';
