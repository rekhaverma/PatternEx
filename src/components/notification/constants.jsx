export const notificationConstants = {
  popupError: 'You need to enable pop-ups on your browser!',
};
