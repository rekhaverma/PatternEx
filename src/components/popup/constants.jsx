export const NOCSVDATA = 'noCSVData';
export const SETLABELSUCCESS = 'setLabelSucces';
export const SETLABELFAILED = 'setLabelFailed';
export const LABELSPOPUP = 'labelPopup';
