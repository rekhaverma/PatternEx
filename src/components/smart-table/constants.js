export const sizePerPageTable = [
  {
    value: '5 results',
    id: '5',
  }, {
    value: '10 results',
    id: '10',
  }, {
    value: '20 results',
    id: '20',
  }, {
    value: '50 results',
    id: '50',
  }, {
    value: '75 results',
    id: '75',
  }, {
    value: '100 results',
    id: '100',
  },
];

export const sortTypes = {
  ASC: 'ASC',
  DESC: 'DESC',
};
