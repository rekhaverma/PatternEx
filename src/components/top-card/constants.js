export const MALICIOUS = 'malicious';
export const SUSPICIOUS = 'suspicious';
export const PRECISIONS = 'precision';
export const LABELS = 'price-tag';
export const REPORT = 'ipo';
