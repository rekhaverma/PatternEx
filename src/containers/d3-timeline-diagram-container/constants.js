import moment from 'moment';

export const defaultStartDate = moment().subtract(7, 'day');
export const defaultEndDate = moment();
