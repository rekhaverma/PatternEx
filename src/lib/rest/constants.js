import axios from 'axios';
import { api } from '../../config';

export const ptrxREST = axios.create({
  'baseURL': `${api.base}/${api.version}`,
  'timeout': 50000,
  'withCredentials': api.withCredentials,
  'headers': {
    'Content-Type': 'application/json',
  },
});

export const ptrxRESTFormData = axios.create({
  'baseURL': `${api.base}/${api.version}`,
  'timeout': 50000,
  'withCredentials': api.withCredentials,
  'headers': {
    'Content-Type': 'application/x-www-form-urlencoded; charset=UTF-8',
    'X-Requested-With': 'XMLHttpRequest',
  },
});

export const ptrxZeppelin = axios.create({
  'baseURL': `${api.zeppelin}`,
  'timeout': 50000,
  'withCredentials': api.withCredentials,
  'headers': {
    'Content-Type': 'application/json',
  },
});

export const ptrxAmbari = axios.create({
  'baseURL': `${api.ambari}/${api.version}`,
  'timeout': 50000,
  'withCredentials': api.withCredentials,
  'headers': {
    'Content-Type': 'application/json',
  },
});

export const ptrxRESTlocal = axios.create({
  'baseURL': process.env.NODE_ENV === 'production' ? '/' : 'https://localhost:8443',
  'timeout': 50000,
  'withCredentials': api.withCredentials,
  'followAllRedirects': true,
  'headers': {
    'Content-Type': 'application/json',
  },
});
