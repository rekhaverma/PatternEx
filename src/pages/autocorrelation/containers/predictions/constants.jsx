import React from 'react';
import { mapIcons } from 'lib';

import { pipelineToName } from 'lib/decorators';
import Tooltip from 'components/tooltip';

export const predictionsColumns = [
  {
    'className': 'table__headerColumn',
    'isKey': true,
    'field': 'hash',
    'hidden': true,
    'label': 'Hash',
    'width': '0px',
  },
  {
    'className': 'table__headerColumn',
    'isKey': false,
    'field': 'momentDay',
    'hidden': false,
    'label': 'Date',
    'width': '100px',
  },
  {
    'className': 'table__headerColumn--entity',
    'isKey': false,
    'field': 'entity_name',
    'hidden': false,
    'label': 'Entity',
    'dataFormat': (cell, row) => (
      <span className="table__cell +ellipsis">
        <span className="table__predictionEntityCell">
          <div className="table__predictionEntityCell--flex">
            <Tooltip
              trigger={(
                <span className={mapIcons(row.entity_type)} />
              )}
              position="bottom"
              key={row}
            >
              {row.entity_type ? pipelineToName(row.entity_type) : ''}
            </Tooltip>
            {row.is_central_entity &&
              <Tooltip
                trigger={(
                  <span className="icon-M-icon" />
                )}
                position="bottom"
              >
                Central entity
              </Tooltip>
            }
          </div>
        </span>
        {cell}
      </span>
    ),
    'width': '250px',
  },
  {
    'className': 'table__headerColumn--highlighted',
    'isKey': false,
    'field': 'threat',
    'hidden': false,
    'label': 'Threat Tactic',
    'width': '90px',
  },
  {
    'className': 'table__headerColumn',
    'isKey': false,
    'field': 'score',
    'hidden': false,
    'label': 'Score',
    'width': '50px',
  },
  {
    'className': 'table__headerColumn--actions',
    'isKey': false,
    'field': 'id',
    'hidden': false,
    'label': 'Actions',
    'dataFormat': (cell, row) => (
      <span className="table__cell--actions">
        {
          row.behavior &&
          row.behavior.toLowerCase() !== 'unknown' && (
            <Tooltip
              trigger={(
                <span
                  className="icon-tab"
                  onClick={e => row.handlers.openEVP(e, row)}
                />
              )}
              position="bottom"
              key={row}
            >
              See Details
            </Tooltip>
          )
        }
        <span>
          <Tooltip
            trigger={(
              <span
                className="icon-add"
                onClick={e => row.handlers.setLabel(e, row)}
              />
            )}
            position="bottom"
            key={row}
          >
            Add Label
          </Tooltip>
        </span>
      </span>
    ),
    'width': '80px',
  },
];

export const labelsColums = [
  {
    'className': 'table__headerColumn',
    'isKey': true,
    'field': 'hash',
    'hidden': true,
    'label': 'Hash',
    'width': '0px',
  },
  {
    'className': 'table__headerColumn',
    'isKey': false,
    'field': 'momentDay',
    'hidden': false,
    'label': 'Update Date',
    'width': '80px',
  },
  {
    'className': 'table__headerColumn--entity',
    'isKey': false,
    'field': 'entity_name',
    'hidden': false,
    'label': 'Entity',
    'dataFormat': (cell, row) => (
      <span className="table__cell +ellipsis">
        <span className="table__predictionEntityCell">
          <div className="table__predictionEntityCell--flex">
            <Tooltip
              trigger={(
                <span className={mapIcons(row.entity_type)} />
              )}
              position="bottom"
              key={row}
            >
              {row.entity_type ? pipelineToName(row.entity_type) : ''}
            </Tooltip>
            {row.is_central_entity &&
              <Tooltip
                trigger={(
                  <span className="icon-M-icon" />
                )}
                position="bottom"
              >
                Central entity
              </Tooltip>
            }
          </div>
        </span>
        {cell}
      </span>
    ),
    'width': '250px',
  },
  {
    'className': 'table__headerColumn--highlighted',
    'isKey': false,
    'field': 'threat',
    'hidden': false,
    'label': 'Threat Tactic',
    'width': '100px',
  },
  {
    'className': 'table__headerColumn',
    'isKey': false,
    'field': 'action',
    'hidden': false,
    'label': 'Actions',
    'width': '100px',
    'dataFormat': (cell, row) => (
      <span className="table__cell">
        <span className="table__predictionEntityCell--flex +spaceBetween">
          Change Label
          <span className={row.action === true ? 'icon-confirm' : 'icon-deny'} />
        </span>
      </span>
    ),
  },
];
