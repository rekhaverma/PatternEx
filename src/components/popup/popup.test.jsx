import React from 'react';
import Popup from './popup.component';
import { NOCSVDATA, SETLABELFAILED, SETLABELSUCCESS, LABELSPOPUP } from './constants';

/**
 * @status: Complete
 * @sign-off-by: Alex Andries
 */
describe('<Popup />', () => {
  let wrapper;
  let props;
  it('should match with snapshot with version 1', () => {
    props = {
      children: <div className="testing-popup" />,
      version: 1,
      context: SETLABELSUCCESS,
      onClose: jest.fn(),
    };
    wrapper = shallow(<Popup {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should match with snapshot with another version and context "NOCSVDATA"', () => {
    props = {
      children: <div className="testing-popup" />,
      version: 2,
      context: NOCSVDATA,
      onClose: jest.fn(),
    };
    wrapper = shallow(<Popup {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should match with snapshot with another version and context "SETLABELFAILED"', () => {
    props = {
      children: <div className="testing-popup" />,
      version: 2,
      context: SETLABELFAILED,
      onClose: jest.fn(),
    };
    wrapper = shallow(<Popup {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should match with snapshot with another version and context "SETLABELSUCCESS"', () => {
    props = {
      children: <div className="testing-popup" />,
      version: 2,
      context: SETLABELSUCCESS,
      onClose: jest.fn(),
    };
    wrapper = shallow(<Popup {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should match with snapshot with another version and context "LABELSPOPUP"', () => {
    props = {
      children: <div className="testing-popup" />,
      version: 2,
      context: LABELSPOPUP,
      onClose: jest.fn(),
    };
    wrapper = shallow(<Popup {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should call onClose when element is clicked', () => {
    props = {
      children: <div className="testing-popup" />,
      version: 1,
      context: SETLABELSUCCESS,
      onClose: jest.fn(),
    };
    wrapper = shallow(<Popup {...props} />);
    wrapper.find('.icon-close').simulate('click');
    expect(props.onClose.mock.calls.length).toBe(1);
  });
});
