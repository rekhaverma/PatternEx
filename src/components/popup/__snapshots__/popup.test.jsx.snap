// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Popup /> should match with snapshot with another version and context "LABELSPOPUP" 1`] = `
<DynamicPopup
  className="popupV2"
  onClick={[Function]}
  text="No Data Found"
/>
`;

exports[`<Popup /> should match with snapshot with another version and context "NOCSVDATA" 1`] = `
<Connect(NoCSVData)
  className="popupV2"
  text="No data found."
/>
`;

exports[`<Popup /> should match with snapshot with another version and context "SETLABELFAILED" 1`] = `
<Connect(NoCSVData)
  className="popupV2"
  text={
    <span
      style={
        Object {
          "textAlign": "center",
        }
      }
    >
      <p>
        Set label failed.
      </p>
      <p>
        Please contact the admin.
      </p>
    </span>
  }
/>
`;

exports[`<Popup /> should match with snapshot with another version and context "SETLABELSUCCESS" 1`] = `
<Connect(NoCSVData)
  className="popupV2"
  text={
    <span
      style={
        Object {
          "textAlign": "center",
        }
      }
    >
      <p>
        Set label succeded.
      </p>
    </span>
  }
/>
`;

exports[`<Popup /> should match with snapshot with version 1 1`] = `
<div
  className="popup"
>
  <div
    className="popup__wrapper"
    style={Object {}}
  >
    <div
      className="popup__header"
    >
      <span
        className="icon-close"
        onClick={[Function]}
      />
    </div>
    <div
      className="popup__content"
    >
      <div
        className="testing-popup"
      />
    </div>
  </div>
  <div
    className="popup__overlay"
    onClick={[Function]}
  />
</div>
`;
